mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala

```diff
@@ -175,6 +177,33 @@ class VectorsSuite extends FunSuite {
     assert(v.size === x.rows)
   }
 
+  test("sqdist") {
+    val random = new Random(System.nanoTime())
+    for (m <- 1 until 1000 by 10) {
+      val nnz = random.nextInt(m) + 1
+
+      val indices1 = random.shuffle(0 to m - 1).toArray.slice(0, nnz).sorted
```

Comment:
Interesting, I tried `shuffle(0 until m)` but it didn't work. Is it a known Scala bug?

I suggest `random.shuffle(0 to m - 1).slice(0, nnz).sorted.toArray` so people can tell the type easily.